qa/acceptance-tests/src/test/java/io/camunda/it/auditlog/AbstractAuditLogClientIT.java

```diff
@@ -0,0 +1,152 @@
+              // store one audit log key for later tests
+              auditLogKey = auditLogItems.items().get(0).getAuditLogKey();
+            });
+  }
+
+  @Test
+  void shouldCreateAuditLogEntryFromPICreation(
+      @Authenticated(DEFAULT_USERNAME) final CamundaClient adminClient) {
+    // given
+    final var processInstanceKey = processInstanceEvent.getProcessInstanceKey();
+
+    // when
+    final var auditLogItems =
+        adminClient
+            .newAuditLogSearchRequest()
+            .filter(
+                fn ->
+                    fn.processInstanceKey(String.valueOf(processInstanceKey))
+                        .operationType(AuditLogOperationTypeEnum.CREATE))
+            .send()
+            .join();
+
+    // then
+    final var auditLog = auditLogItems.items().get(0);
+    assertThat(auditLog).isNotNull();
+    assertThat(auditLog.getEntityType()).isEqualTo(AuditLogEntityTypeEnum.PROCESS_INSTANCE);
+    assertThat(auditLog.getOperationType()).isEqualTo(AuditLogOperationTypeEnum.CREATE);
+    assertThat(auditLog.getCategory()).isEqualTo(AuditLogCategoryEnum.DEPLOYED_RESOURCES);
+    assertThat(auditLog.getProcessDefinitionId())
+        .isEqualTo(processInstanceEvent.getBpmnProcessId());
+    assertThat(auditLog.getProcessDefinitionKey())
+        .isEqualTo(String.valueOf(processInstanceEvent.getProcessDefinitionKey()));
+    assertThat(auditLog.getProcessInstanceKey())
+        .isEqualTo(String.valueOf(processInstanceEvent.getProcessInstanceKey()));
+    assertThat(auditLog.getTenantId()).isEqualTo(TENANT_A);
+    assertThat(auditLog.getResult()).isEqualTo(AuditLogResultEnum.SUCCESS);
+  }
+
+  @Test
+  void shouldFetchAuditLogEntryByAuditLogKey(
+      @Authenticated(DEFAULT_USERNAME) final CamundaClient adminClient) {
+    // when
+    final var fetchedAuditLog = adminClient.newAuditLogGetRequest(auditLogKey).send().join();
+
+    // then
+    // verify the fetched audit log entry matches the searched one
+    assertThat(fetchedAuditLog).isNotNull();
+    assertThat(fetchedAuditLog.getAuditLogKey()).isEqualTo(auditLogKey);
+    assertThat(fetchedAuditLog.getEntityType()).isEqualTo(AuditLogEntityTypeEnum.PROCESS_INSTANCE);
+    assertThat(fetchedAuditLog.getOperationType()).isEqualTo(AuditLogOperationTypeEnum.CREATE);
+    assertThat(fetchedAuditLog.getCategory()).isEqualTo(AuditLogCategoryEnum.DEPLOYED_RESOURCES);
+    assertThat(fetchedAuditLog.getProcessDefinitionId())
+        .isEqualTo(processInstanceEvent.getBpmnProcessId());
+    assertThat(fetchedAuditLog.getProcessDefinitionKey())
+        .isEqualTo(String.valueOf(processInstanceEvent.getProcessDefinitionKey()));
+    assertThat(fetchedAuditLog.getProcessInstanceKey())
+        .isEqualTo(String.valueOf(processInstanceEvent.getProcessInstanceKey()));
+    assertThat(fetchedAuditLog.getTenantId()).isEqualTo(TENANT_A);
+    assertThat(fetchedAuditLog.getResult()).isEqualTo(AuditLogResultEnum.SUCCESS);
+  }
```

Comment:
We test the `creation` and `getByKey` here, why not the search as well? What is our testing concept/purpose/goal here?

Reply:
We test the search API, not `creation`. The test concept for the `AuditLog[Rest\GRPC]ClientIT` tests is to cover the Java Client and REST API endpoints.